controllers/servers/host_definer/watcher/watcher_helper.py

```diff
@@ -96,22 +97,30 @@ def _delete_definition(self, host_definition):
         host_definition_instance, _ = self._get_host_definition(host_definition.node_name, host_definition.secret)
         if response.error_message and host_definition_instance:
             self._set_host_definition_status(host_definition_instance.name, settings.PENDING_DELETION_PHASE)
-            self._create_event_to_host_definition(host_definition_instance, response.error_message)
+            self._create_fail_event_for_host_definition(
+                host_definition_instance, response.error_message, settings.UNDEFINE)
         elif not response.error_message and host_definition_instance:
             self._delete_host_definition(host_definition_instance.name)
             self._remove_managed_by_host_definer_label(node_name)
 
     def _undefine_host(self, host_definition):
         return self._ensure_definition_state(host_definition, self.storage_host_servicer.undefine_host)
 
-    def _create_event_to_host_definition(self, host_definition, message):
-        self._add_event_to_host_definition(host_definition, message)
-        if host_definition:
-            self._add_event_to_host_definition(host_definition, message)
+    def _create_fail_event_for_host_definition(self, host_definition, message, action):
+        logger.info(messages.CREATE_ERROR_EVENT_FOR_HOST_DEFINITION.format(host_definition.name, message))
+        self._generate_and_create_event(host_definition, action, message, settings.FAILED)
+
+    def _set_host_definition_status_to_ready(self, host_definition):
+        self._set_host_definition_status(host_definition.name, settings.READY_PHASE)
+        self._create_success_event_for_host_definition(host_definition, settings.DEFINE)
+
+    def _create_success_event_for_host_definition(self, host_definition, action):
+        message = settings.SUCCESS_MESSAGE
+        logger.info(messages.CREATE_SUCCESS_EVENT_FOR_HOST_DEFINITION.format(host_definition.name))
+        self._generate_and_create_event(host_definition, action, message, settings.SUCCESSFUL)
 
-    def _add_event_to_host_definition(self, host_definition, message):
-        logger.info(messages.CREATE_EVENT_FOR_HOST_DEFINITION.format(host_definition.name, message))
-        event = self._get_event_for_host_definition(host_definition, message)
+    def _generate_and_create_event(self, host_definition, action, message, message_type):
```

Comment:
cant see why it's a better name

Reply:
better name from what?

Reply:
for `_add_event_to_host_definition`

Reply:
I don't know why GitHub showed that it changed `_add_event_to_host_definition`, but `_generate_and_create_event` function doesn't replace `_add_event_to_host_definition`

Reply:
I can't see the diff in the 2 functions' logic.. 
anyway, please consider
```suggestion
    def _create_host_definition_event(self, host_definition, action, message, message_type):
```

Reply:
> I can't see the diff in the 2 functions' logic.. anyway, please consider
you can see that `_add_event_to_host_definition` was converted to two function: `_create_success_event_for_host_definition` and `_create_fail_event_for_host_definition` and there was a code dup between them so I created `_generate_and_create_event` function

Reply:
done as we discussed in the Webex conversation with @ELENAGER